### PR TITLE
Fix ragged_tensor resize new case

### PR DESCRIPTION
**Pull request by bhack**
# What does this PR do?

Fixes  https://github.com/keras-team/keras-cv/issues/841#issuecomment-1269481988

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [X] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@martin-gorner @LukeWood

### CODE REVIEW

**Comment by bhack** [comment]
@martin-gorner 
Your example is at https://colab.research.google.com/gist/bhack/eb1372d4749851dbc33a20e3d8485ef3/test_ragged_resize.ipynb

**Comment by bhack** [comment]
Do you have a priority for this?

**Review by tanzhenyu — COMMENTED** [review]
Thanks for the PR!

File: keras_cv/bounding_box/converters.py [hunk 1]
```diff
@@ -373,6 +373,9 @@ def _image_shape(images, image_shape, boxes):
         else:
             height = tf.reshape(images.row_lengths(), (-1, 1))
             width = tf.reshape(tf.reduce_max(images.row_lengths(axis=2), 1), (-1, 1))
+            if isinstance(boxes, tf.RaggedTensor):
```
[the remark this answers is not in the document]
Comment (bhack) [reply keras_cv/bounding_box/converters.py:376]: The new case will require an heavy refactoring of the tests as discussed in the related ticket. It is why I've exposed a colab as a proxy test.

I think we could do this when we will find a standardized approach in tests to verify ragged and unragged tests at the same time without duplicating too much things.

If you have any quick design workaround I could try to add the Colab case in tests.

[the remark this answers is not in the document]
Comment (LukeWood) [reply keras_cv/bounding_box/converters.py:376]: I've tried rebasing my branch https://github.com/keras-team/keras-cv/pull/1008 on this and it still does not fix the issues unfortunately.

Error report:

```

    in user code:

        File "/Users/lukewood/workspace/keras-cv/keras_cv/layers/preprocessing/base_image_augmentation_layer.py", line 343, in call  *
            return self._format_output(self._batch_augment(inputs), metadata)
        File "/Users/lukewood/workspace/keras-cv/keras_cv/layers/preprocessing/base_image_augmentation_layer.py", line 424, in _batch_augment  *
            return self._map_fn(self._augment, inputs)
        File "/Users/lukewood/workspace/keras-cv/keras_cv/layers/preprocessing/base_image_augmentation_layer.py", line 212, in _map_fn  *
            func, inputs, fn_output_signature=self._compute_output_signature(inputs)
        File "/Users/lukewood/workspace/keras-cv/keras_cv/layers/preprocessing/base_image_augmentation_layer.py", line 392, in _augment  *
            bounding_boxes = self.augment_bounding_boxes(
        File "/Users/lukewood/workspace/keras-cv/keras_cv/layers/preprocessing/random_shear.py", line 165, in augment_bounding_boxes  *
            bounding_boxes = keras_cv.bounding_box.convert_format(
        File "/Users/lukewood/workspace/keras-cv/keras_cv/bounding_box/converters.py", line 384, in convert_format  *
            result = from_xyxy_fn(in_xyxy, images=images, image_shape=image_shape)
        File "/Users/lukewood/workspace/keras-cv/keras_cv/bounding_box/converters.py", line 212, in _xyxy_to_rel_xyxy  *
            left, top, right, bottom, rest = tf.split(

        TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'


    Call arguments received by layer "random_shear" (type RandomShear):
      • inputs={'images': 'tf.RaggedTensor(values=tf.RaggedTensor(values=Tensor("augmenter/random_flip/map/RaggedFromVariant_1/RaggedTensorFromVariant:2", shape=(None, 3), dtype=float32), row_splits=Tensor("augmenter/random_flip/map/RaggedFromVariant_1/RaggedTensorFromVariant:1", shape=(None,), dtype=int64)), row_splits=Tensor("augmenter/random_flip/map/RaggedFromVariant_1/RaggedTensorFromVariant:0", shape=(10,), dtype=int64))', 'bounding_boxes': 'tf.RaggedTensor(values=tf.RaggedTensor(values=Tensor("augmenter/random_flip/map/RaggedFromVariant/RaggedTensorFromVariant:2", shape=(None,), dtype=float32), row_splits=Tensor("augmenter/random_flip/map/RaggedFromVariant/RaggedTensorFromVariant:1", shape=(None,), dtype=int64)), row_splits=Tensor("augmenter/random_flip/map/RaggedFromVariant/RaggedTensorFromVariant:0", shape=(10,), dtype=int64))'}
      • training=True
```

[the remark this answers is not in the document]
Comment (bhack) [reply keras_cv/bounding_box/converters.py:376]: What issue? As I have exactly replicated the original use case in the connected ticket with the Colab https://colab.research.google.com/gist/bhack/eb1372d4749851dbc33a20e3d8485ef3/test_ragged_resize.ipynb

Do we have a new user case?

[the remark this answers is not in the document]
Comment (LukeWood) [reply keras_cv/bounding_box/converters.py:376]: Scratch that; I fixed my case and it was actually incorrect.

**Review by LukeWood — APPROVED** [review]

File: keras_cv/bounding_box/converters.py [hunk 2]
```diff
@@ -373,6 +373,9 @@ def _image_shape(images, image_shape, boxes):
...
+                height = tf.expand_dims(height, axis=-1)
```
Comment (LukeWood) [inline keras_cv/bounding_box/converters.py:377]: @bhack can you add a comment explaining why this works?

Comment (bhack) [reply keras_cv/bounding_box/converters.py:377]: It was more then 1 month ago but if I remember correctly is that we need to add an extra dim in the ragged case as we are using high and width as multipliers in rel conversion.

File: keras_cv/bounding_box/converters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bhack) [reply keras_cv/bounding_box/converters.py:376]: > Scratch that; I fixed my case and it was actually incorrect.

Just as a reminder and also not related to this ticket:
If I remember correctly bbox conversion are not supporting ragged tensor in graph (autograph) mode as in the impl we access to the `.shape` of the boxes.

**Comment by martin-gorner** [comment]
> If I remember correctly bbox conversion are not supporting ragged tensor in graph (autograph) mode as in the impl we access to the .shape of the boxes

We'll have to see if that's a problem on some platforms (TPU?). Maybe not but testing is needed. Thank you @bhack for flagging this.

**Comment by bhack** [comment]
Yes as it is not strictly connected with this this fix and its related issue we could eventually open a new ticket.

**Review by qlzh727 — APPROVED** [review]
Thanks for the fix.

**Comment by bhack** [comment]
The connected issue was closed if you don't need this I will close as it is already 2 months old now.

**Comment by qlzh727** [comment]
Not sure why the issue is closed, but I think this fix is needed for the issue. I will merge it now.

**Pull request merged**